jetbrains-core/src/software/aws/toolkits/jetbrains/ui/feedback/FeedbackDialog.kt

```diff
@@ -32,14 +38,20 @@ class FeedbackDialog(project: Project) : DialogWrapper(project) {
 
             val sentiment = panel.sentiment ?: throw IllegalStateException("sentiment was null after validation")
             val comment = panel.comment ?: throw IllegalStateException("comment was null after validation")
-            ApplicationManager.getApplication().executeOnPooledThread {
+            GlobalScope.launch(ApplicationThreadPool) {
```

Comment:
I *think* you want to only group the actions that can safely be cancelled together under a single non-supervisor scope.

And group all actions tied to your application scope to an uber supervisor scope.

I believe GlobalScope is non-supervisor, which means if any of the coroutines are cancelled here, it will cancel all of them which doesn't seem correct.

So at the least, we should create our own supervisor scope and launch coroutines in it. Then we can safely create `coroutineScope { }` stuff to group the actions that can safely be cancelled together without cancelling stuff outside of their scope.

Reply:
This is what I'm thinking:

* Instead of `GlobalScope`, create a new scope inside of `FeedbackDialog`
  * `val job = SupervisorJob()`
  * `val scope = job + ApplicationThreadPool + CoroutineName("FeedbackDialog")`

* Change call to `GlobalScope.launch` in `FeedbackDialog` to `scope.launch`

* Change `TelemetryService.sendFeedback` to be `suspend`, since we control it and can say it's a suspendable operation.

* Remove the `runBlocking` from `TelemetryService.sendFeedback`, since it's suspending and we can control the blocking behavior in the caller. Since, in your current implementation, the caller is not doing async stuff, it will be synchronous by default.

Reply:
should we have a lint rule that stops us from using global scope?

Reply:
spun up a new class that implements `CoroutineScope` and had `FeedbackDialog` delegate out `CoroutineScope` to that

+1 for globalscope lint, not in this PR though.